### PR TITLE
[distributions] Fix incorrect mode formula in Kumaraswamy distribution

### LINKED ISSUE #173912: Wrong mode in `torch.distributions.Kumaraswamy`
### 🐛 Describe the bug

The current implementation of the mode in `torch.distributions.Kumaraswamy` appears to be implemented in a mathematically incorrect way. The definition of the mode is 

$$
\text{Mode}(X) = \left( \frac{a - 1}{a b  - 1} \right)^{\frac1a}, \quad X \sim \text{Kumaraswamy}(a, b).
$$ 

The current implementation, as far as I can tell, is instead computing

$$
\frac{(1 - b)^{\frac1b}}{1 - a b},
$$

resulting in outputs being almost always NaNs (or just wrong). You can reproduce the problem with the following minimal example:

```python
import torch
from torch.distributions import Kumaraswamy

torch.set_default_dtype(torch.float64)

# define some test cases with different shapes
shapes = torch.tensor(
    [
        [0.5, 0.5],  # expected nan
        [1.0, 0.5],  # expected nan
        [0.5, 1.0],  # expected nan
        [1.0, 1.0],  # expected nan
        [3.0, 1.0],  # expected 1.0
        [1.0, 3.0],  # expected 0.0
        [3.0, 5.0],  # expected 0.522...
        [1.000000001, 1.0000000000001],  # expected 0.999...
        [2.0, 1e19],  # expected 0.00000000022360679774997...
    ]
)
expected_mode = torch.tensor(
    [
        torch.nan,
        torch.nan,
        torch.nan,
        torch.nan,
        1.0,
        0.0,
        0.52275795857471021674829618715991546621,
        0.99990000999900010498883452487251366936,
        0.00000000022360679774997896964650753681,
    ]
)
a, b = shapes.T

# compute the desired mode
desired_mode = ((a - 1) / (a * b - 1)).pow(a.reciprocal())
desired_mode[(a < 1) | (b < 1) | ((a == 1) & (b == 1))] = torch.nan

# compute the mode from the official implementation
distr = Kumaraswamy(a, b)
actual_mode = distr.mode

torch.testing.assert_close(desired_mode, expected_mode, equal_nan=True)  # passes
torch.testing.assert_close(actual_mode, expected_mode, equal_nan=True)   # fails
```

I'd be happy to work on a PR amending this issue and to add new test cases for the Kumaraswamy mode.

---

PS: I am dropping here alternative implementations of the mode computation in log-space 
```python
desired_mode = (a.reciprocal() * ((a - 1).log() - (a * b - 1).log())).exp()
desired_mode = (a.reciprocal() * ((a - 2).log1p() - (a * b - 2).log1p())).exp()
``` 

Small numerical tests appear to show that the implementation with `.pow` (i.e., not in log space) actually yields lower numerical errors than the log implementations. This is just an empirical guess and no definite conclusion. However, the log implementations enjoy the benefit of yielding `nan`'s in the exact spots without having to manually mask for entries in which $a<1, b<1, a=b=1$. Issue pytorch/pytorch#139019 might be related.

### Versions

Collecting environment information...
PyTorch version: 2.10.0+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version: 18.1.3 (1ubuntu1)
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.13.5 | packaged by conda-forge | (main, Jun 16 2025, 08:27:50) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.14.0-37-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           43 bits physical, 48 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  128
On-line CPU(s) list:                     0-127
Vendor ID:                               AuthenticAMD
Model name:                              AMD Ryzen Threadripper PRO 3995WX 64-Cores
CPU family:                              23
Model:                                   49
Thread(s) per core:                      2
Core(s) per socket:                      64
Socket(s):                               1
Stepping:                                0
Frequency boost:                         enabled
CPU(s) scaling MHz:                      55%
CPU max MHz:                             4309.0000
CPU min MHz:                             550.0000
BogoMIPS:                                5389.91
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba ibpb stibp vmmcall fsgsbase bmi1 avx2 smep bmi2 cqm rdt_a rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local clzero irperf xsaveerptr rdpru wbnoinvd arat npt lbrv svm_lock nrip_save tsc_scale vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif v_spec_ctrl umip rdpid overflow_recov succor smca sev sev_es
Virtualization:                          AMD-V
L1d cache:                               2 MiB (64 instances)
L1i cache:                               2 MiB (64 instances)
L2 cache:                                32 MiB (64 instances)
L3 cache:                                256 MiB (16 instances)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-127
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Mitigation; untrained return thunk; SMT enabled with STIBP protection
Vulnerability Spec rstack overflow:      Mitigation; Safe RET
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Retpolines; IBPB conditional; STIBP always-on; RSB filling; PBRSB-eIBRS Not affected; BHI Not affected
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

Versions of relevant libraries:
[pip3] numpy==2.4.1
[pip3] torch==2.10.0+cpu
[conda] numpy                     2.4.1                    pypi_0    pypi
[conda] torch                     2.10.0+cpu               pypi_0    pypi

cc @fritzo @neerajprad @alicanb @nikitaved

### PR DESCRIPTION
**Pull request by subinz1**
## Summary

Fixes the `mode` property of the `Kumaraswamy` distribution which was using a mathematically incorrect formula.

## Changes

**The correct formula for the mode is:**
$$\text{Mode}(X) = \left( \frac{a - 1}{a b - 1} \right)^{\frac{1}{a}}, \quad X \sim \text{Kumaraswamy}(a, b)$$

**The previous implementation was computing:**
$$\frac{(1 - b)^{\frac{1}{b}}}{1 - ab}$$

This resulted in incorrect values (or NaN) for most inputs.

**The mode is undefined (NaN) when:**
- `a < 1` (distribution is U-shaped at lower boundary)
- `b < 1` (distribution is U-shaped at upper boundary)
- `a = b = 1` (uniform distribution, no unique mode)

## Test plan

Added `test_kumaraswamy_mode` test with the following test cases:
| concentration1 (a) | concentration0 (b) | Expected mode |
|---|---|---|
| 0.5 | 0.5 | NaN (a < 1 and b < 1) |
| 1.0 | 0.5 | NaN (b < 1) |
| 0.5 | 1.0 | NaN (a < 1) |
| 1.0 | 1.0 | NaN (a = b = 1) |
| 3.0 | 1.0 | 1.0 (mode at upper boundary) |
| 1.0 | 3.0 | 0.0 (mode at lower boundary) |
| 3.0 | 5.0 | 0.5228... |

## Reproduction

```python
import torch
from torch.distributions import Kumaraswamy

# Before fix: returns NaN or wrong values
# After fix: returns correct mode
dist = Kumaraswamy(torch.tensor(3.0), torch.tensor(5.0))
print(dist.mode)  # Expected: 0.5228...
```

Fixes #173912

cc @fritzo @neerajprad @alicanb @nikitaved

### CODE REVIEW

**Comment by albanD** [comment]
non-actionable issue, see comment on your other PR
